.github/workflows/build-server.yaml

```
@@ -130,56 +130,92 @@ jobs:
+            -t $AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$CONTAINER_IMAGE-demo:${{ github.sha }} \
+            -t $CACHE_IMAGE \
+            .
+
+          # Push both main image and cache
           docker push $AWS_ACCOUNT_ID.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$CONTAINER_IMAGE-demo:${{ github.sha }}
+          docker push $CACHE_IMAGE
 
-      - name: Push (PROD)
+      - name: Build and push image (PROD) - With Registry Caching
         if: github.ref == 'refs/heads/main'
         run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:2:37: Double quote to prevent globbing and word splitting

(shellcheck)